Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -410,9 +412,43 @@
     return false;
 }
 
-bool UnifiedPDFPlugin::handleContextMenuEvent(const WebMouseEvent&)
+PDFContextMenu UnifiedPDFPlugin::createContextMenu(const IntPoint& contextMenuPoint) const
 {
-    return false;
+    Vector<PDFContextMenuItem> menuItems;
+
+    // FIXME: We should also set the openInPreviewIndex when UnifiedPdfPlugin::openWithPreview is implemented
+    menuItems.append({ WebCore::contextMenuItemPDFOpenWithPreview(), true, false, 0, true, static_cast<uint8_t>(ContextMenuAction::OpenWithPreview) });
+
+    return { contextMenuPoint, WTFMove(menuItems), { } };
+}
+
+bool UnifiedPDFPlugin::handleContextMenuEvent(const WebMouseEvent& event)
+{
+    if (!m_frame || !m_frame->coreLocalFrame())
+        return false;
+    RefPtr webPage = m_frame->page();
+    if (!webPage)
+        return false;
+    RefPtr frameView = m_frame->coreLocalFrame()->view();
+    if (!frameView)
+        return false;
+
+    auto contextMenu = createContextMenu(frameView->contentsToScreen(IntRect(frameView->windowToContents(event.position()), IntSize())).location());
+
+    auto sendResult = webPage->sendSync(Messages::WebPageProxy::ShowPDFContextMenu(contextMenu, m_identifier));
+    auto [selectedItemTag] = sendResult.takeReplyOr(-1);
+
+    if (selectedItemTag && *selectedItemTag >= 0 && static_cast<unsigned>(*selectedItemTag) < contextMenu.items.size())
```

Comment:
It's confusing that you use the term "Tag" yet `ContextMenuAction` doesn't have "tag" in the name.

Also I don't think you should assume that tags equate to indexes. Tags might start at 1000 but this code should still work.

Reply:
Ended up renaming to `ContextMenuItemTag`